### ISSUE TITLE
layoutSubviews swizzling causes an infinite loop 

### ISSUE THREAD

**Issue by artgoncharov** [Juanpe/SkeletonView#495]
### Description

layoutSubviews swizzling causes issues for UIKit. 

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/Juanpe/SkeletonView/blob/main/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://github.com/Juanpe/SkeletonView/blob/main/CODE_OF_CONDUCT.md).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### SkeletonView Environment:

**SkeletonView version: 1.29.2**
**Xcode version: 13.2.1**
**Swift version: 5**

#### Steps to reproduce:

1. Setup the floating keyboard on iPad
2. Open app in a slide over mode
3. Show a skeleton animation 
4. Focus on a textfield with a toolbar

#### Expected result:

Keyboard is shown

#### Actual result:

App freezes in the infinite loop

#### Attachments:

Logs, screenshots, sample project, funny gif, etc.

https://user-images.githubusercontent.com/50835041/159222726-e042e8e0-cd97-43e0-9f01-85c98e54771b.mov

[Sample project](https://github.com/Juanpe/SkeletonView/files/8313823/SkeletonSwizzlingFailure.zip)

**Comment by check4ismail** [comment]
Hey @artgoncharov ,

Would you be able to include your code on here? If not, feel free to share code that mimics the issue. It'll help further narrow down the issue to figure out what's going on. Thanks!

**Comment by artgoncharov** [comment]
Hello @check4ismail I've attached the sample project in the description. I will dup the link here [Sample project](https://github.com/Juanpe/SkeletonView/files/8313823/SkeletonSwizzlingFailure.zip)

**Comment by check4ismail** [comment]
Hey @artgoncharov ,

Here's what I've done so far:
- Downloaded your sample project
- Instead of using Cocoapods, went with Swift Package Manager (my preferred package manager)
- Ran the app using the an iPad Pro 11-inch (3rd generation) simulator running iOS 15.2

I wasn't able to replicate the issue and was able to pull up the keyboard as expected from Textfield after tapping the **Toggle Skeleton** button. Is there anything else I'm missing here just to make sure we're on the same page?

**Comment by artgoncharov** [comment]
@check4ismail just to make sure, did you set up the floating keyboard before? Also, run the app in slide over mode (check the vid). BTW, you can use _Reproduce Infinite Loop_ button. Thank you in advance!

**Comment by check4ismail** [comment]
@artgoncharov ,

Appreciate you mentioning those things as well. And correct, I was able to do the following at the same time: 1) type onto the Textfield using a floating keyboard, 2) make sure the app is running in slide over mode

If you're curious about the sequence of events here it is below:
- Run app
- Change app view mode to slide over mode
- Change keyboard to floating
- Tap on `Reproduce Infinite Loop` button
- From there, I was able to type into the Textfield without an issue

**Comment by artgoncharov** [comment]
I think we need someone else to try. I just tried to reproduce on a fresh install and I'm still able to reproduce.

**Comment by Juanpe** [comment]
Hi! I've downloaded the project and I can reproduce this issue. I'm not exactly sure what it's happening 🤔 

The steps to reproduce it are:
- Run app
- Change app view mode to slide over mode
- Change keyboard to floating
- Toggle skeleton
- When skeleton is active, tap on the text field. 
- The memory starts to increase exponentially.

https://user-images.githubusercontent.com/1409041/159485994-b0e9f990-dfc4-480a-a876-183228ac7f9a.mov

**Comment by artgoncharov** [comment]
@Juanpe it's an infinite loop :) if you remove this swizzling https://github.com/Juanpe/SkeletonView/blob/main/SkeletonViewCore/Sources/Internal/UIKitExtensions/UIView%2BSwizzling.swift#L34 you won't get this issue

**Comment by Juanpe** [comment]
Yeah, exactly, but the question is what is the cause of this infinite loop 😅 I can see that when the keyboard appears, some constraints are broken and it seems could be the cause 🤔

**Comment by JackTheTapper** [comment]
Hi, any news on this? Im seeing the same thing, when updating collectionviews and am eager to hear how we can aviod this loop

**Comment by drash-course** [comment]
I'm facing the same problem (version 1.8.2)

**Comment by ihnatmoisieiev** [comment]
Same issue. Any updates here?

**Comment by MashfiqueA** [comment]
This is currently a trending crash in my app. Any updates on this?

Is this fixed by: https://github.com/Juanpe/SkeletonView/pull/574?
